### ISSUE TITLE
How do I keep vscodevim functionality when switching keyboard language layout?

### ISSUE THREAD

**Issue by cgregurich** [VSCodeVim/Vim#7327]
When I switch to a Russian layout, I want the keybindings to still work. For example, when I want to insert in this layout, technically I'll be sending the key "ш", but on the physical keyboard, it's still the "i" key. How do I accomplish this? It makes it annoying because I have to switch keyboard layouts each time I want to do a vim command.

**Comment by w-cantin** [comment]
Hello,

You might have to remap Russian keys to regular vim commands using the `"vim.normalModeKeyBindings"` command in `settings.json`. Make sure to use the `normalModeKeybindings` and not the "normalModeKeybindingsRecursive" for this type of bind.

For example, to rebind the key `ш` to `i` it would be possible to write the following: 
```json
    "vim.normalModeKeyBindings": [
        { "before": ["ш"], "after": ["i"] },
    ],
```

**Comment by kostabekre** [comment]
> Hello,
> 
> You might have to remap Russian keys to regular vim commands using the `"vim.normalModeKeyBindings"` command in `settings.json`. Make sure to use the `normalModeKeybindings` and not the "normalModeKeybindingsRecursive" for this type of bind.
> 
> For example, to rebind the key `ш` to `i` it would be possible to write the following:
> 
> ```json
>     "vim.normalModeKeyBindings": [
>         { "before": ["ш"], "after": ["i"] },
>     ],
> ```

So we have to do it for every letter? Is there an easier method?

**Comment by Evgenii-Barannik** [comment]
Even if you do it for every letter, for every mode, it still does not really work. Individual keys work: for example  "ц" works like "w", but "в в" does not work as "d d". 

Other solutions i tried, like switching layout on mode change, are halfbaked and useless due to their unreliability — switching layout will not work if you change to VsCode window while with Russian layout. 

Do you have something that works?

**Comment by kostabekre** [comment]
Well, in your system you could set the layout stay the same in the apps you need. [Like here](https://superuser.com/questions/766400/make-language-change-specific-to-window-app-not-whole-system-windows-8-1)
But the easiest solution would be change to vim/neovim because they have an [easy solution](https://vi.stackexchange.com/questions/230/how-can-i-modify-vim-to-work-with-a-different-keyboard-layout) for this problem.

**Comment by Evgenii-Barannik** [comment]
@kostabekre Do you mean that you were able to integrate vscodevim with neovim while fixing this issue?

**Comment by kostabekre** [comment]
No, I switched to neovim. There were other issues that I didn't like in
vscodevim, so I just decided to study neovim

**Comment by vigilancer** [comment]
> Even if you do it for every letter, for every mode, it still does not really work. Individual keys work: for example "ц" works like "w", but "в в" does not work as "d d".
> 
> Other solutions i tried, like switching layout on mode change, are halfbaked and useless due to their unreliability — switching layout will not work if you change to VsCode window while with Russian layout.
> 
> Do you have something that works?

for "в в" to work as "d d" you need to override "vim.operatorPendingModeKeyBindings" too.

For example:

```
  "vim.normalModeKeyBindings": [
    { "before": ["в"], "after": ["d"], },
]
"vim.operatorPendingModeKeyBindings": [
   { "before": ["в"], "after": ["d"], },
]
```

Don't look for easy solution, look for the right one.
